addons/sale/models/sale.py

```diff
@@ -1025,23 +1036,37 @@ def _get_to_invoice_qty(self):
             else:
                 line.qty_to_invoice = 0
 
-    @api.depends('invoice_lines.move_id.state', 'invoice_lines.quantity')
+    @api.depends('product_uom', 'invoice_lines.move_id.state', 'invoice_lines.quantity', 'invoice_lines.product_uom_id')
     def _get_invoice_qty(self):
         """
         Compute the quantity invoiced. If case of a refund, the quantity invoiced is decreased. Note
         that this is the case only if the refund is generated from the SO and that is intentional: if
         a refund made would automatically decrease the invoiced quantity, then there is a risk of reinvoicing
         it automatically, which may not be wanted at all. That's why the refund has to be created from the SO
         """
+        origin_lines = self._origin
+        if origin_lines:
+            self.env.cr.execute('''
+                SELECT
+                    rel.order_line_id as id,
+                    SUM(CASE WHEN am.type = 'out_invoice' THEN aml.quantity / aml_uom.factor * sol_uom.factor ELSE 0 END) as out_invoice,
+                    SUM(CASE WHEN am.type = 'out_refund' THEN aml.quantity / aml_uom.factor * sol_uom.factor ELSE 0 END) as out_refund
```

Comment:
missing a rounding when comparing to the `_compute_quantity` method of `uom.uom`